### PR TITLE
feat: added initial CDC implementation with stream API

### PR DESCRIPTION
**Pull request by mattayres**

### CODE REVIEW

File: server/v1/api.proto [hunk 1]
```diff
@@ -423,6 +423,19 @@ message DatabaseMetadata {
...
+message StreamRequest {
+  string db = 1;
+}
+
+message StreamResponse {
+  repeated StreamChange changes = 1;
+}
+
+message StreamChange {
+  string collectionName = 1;
```
Comment (himank) [inline server/v1/api.proto:435]: can you change it to `collection_name`

File: server/v1/api.proto [hunk 2]
```diff
@@ -423,6 +423,19 @@ message DatabaseMetadata {
 message CollectionMetadata {
 }
 
+message StreamRequest {
+  string db = 1;
```
Comment (himank) [inline server/v1/api.proto:427]: add collection name as well here if the user need to filter.

File: server/v1/api.proto [hunk 3]
```diff
@@ -423,6 +423,19 @@ message DatabaseMetadata {
...
+}
+
+message StreamResponse {
+  repeated StreamChange changes = 1;
+}
+
+message StreamChange {
```
Comment (himank) [inline server/v1/api.proto:434]: let's have resume_token as well? similar to ReadResponse?

File: server/v1/api.proto (hunk 1 above)
Comment (mattayres) [reply server/v1/api.proto]: done

File: server/v1/api.proto (hunk 2 above)
Comment (mattayres) [reply server/v1/api.proto:433]: will do after MVP

File: server/v1/api.proto (hunk 3 above)
Comment (mattayres) [reply server/v1/api.proto:440]: will do after MVP

**Review by himank — APPROVED** [review]

**Pull request merged**